src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    const arr = [...this];
+    let str = '';
+
+    for (let i = 0; i < arr.length; i++) {
+      switch (arr[i]) {
+        case undefined:
+          arr[i] = '';
+          break;
+        case null:
+          arr[i] = '';
+          break;
+      }
```

Comment:
Using a switch statement for only two cases is not optimal. Consider using an if statement instead. Just check if element is not undefined or null and add it to the str only in that case